test/zdtm/static/memfd-secret00.c

```diff
@@ -0,0 +1,80 @@
+#include <stdio.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/syscall.h>
+#include <unistd.h>
+
+#include "zdtmtst.h"
+
+#define SECRET "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+#define SIZE   26
+
+const char *test_doc = "memfd_secret file descriptor";
+const char *test_author = "Dhanuka Warusadura <csx@tuta.io>";
+
+static int _memfd_secret(unsigned int flags)
+{
+#if defined(CONFIG_HAS_MEMFD_SECRET)
```

Comment:
Please use dynamic detection. If `syscall(__NR_memfd_secret, 0)` returns `-ENOSYS` just call `skip()` for the test

Reply:
I agree that we don't need CONFIG_HAS_MEMFD_SECRET. But we also don't need a dynamic detection, we have desc file which will get this information from kerndat.

Reply:
Implemented the `desc` file for test.